src/os/bluestore/BlueStore.cc

```diff
@@ -5630,7 +5630,7 @@ int BlueStore::allocate_bluefs_freespace(
 	      << " available 0x " << alloc->get_free()
 	      << std::dec << dendl;
 
-	alloc->dump();
+	_dump_alloc_on_failure();
 	alloc->release(*extents);
```

Comment:
I think that if we do not have entire size we wanted, it is still OK to keep already allocated data.
At least it will clear condition that forced us to gift space bluestore->bluefs.

Reply:
@aclamk - it actually depends on the caller context - sometimes we can proceed with the allocation lower than required - sometimes we can't.